### PR TITLE
vpnc: change upstream to fork

### LINKED ISSUE #26798: vpnc(musl version) cannot connect
System: x86_64 (musl version)
Package: vpnc version 0.5.3

vpnc cannot connect:
When run it says: "can't initialise tunnel interface: No such file or directory".
/dev/net/tun present in system, but vpnc try find /dev/tun0-254 devices

### PR DESCRIPTION
**Pull request by Gottox**
fixes #26798.

* Changes upstream to https://github.com/streambinder/vpnc/. That's a fork
  that's a fork which is at least used in alpine.
* Depends on vpnc-scripts. New upstream doesn't ship those files anymore
* Add wrapper script to stay backwards compatible
* Switch to build_style=gnu-makefile instead of manually calling make.

### CODE REVIEW

**Comment by ericonr** [comment]
Does this also fix https://github.com/void-linux/void-packages/issues/12507 ?

File: srcpkgs/vpnc/template [hunk 1]
```diff
@@ -20,14 +25,16 @@ if [ "$CROSS_BUILD" ]; then
 	hostmakedepends+=" vpnc"
 fi
 
-do_configure() {
+pre_configure() {
```
Comment (ericonr) [inline srcpkgs/vpnc/template:28]: `post_patch`?

File: srcpkgs/vpnc/template [hunk 2]
```diff
@@ -20,14 +25,16 @@ if [ "$CROSS_BUILD" ]; then
...
+	vsed -e 's/^#OPENSSL/OPENSSL/' -i Makefile
+	vsed -e 's/^install:.*/install: install-common/' -i Makefile
+	vsed -e 's#^SBINDIR=.*#SBINDIR=/$(PREFIX)/bin#' -i Makefile
 	if [ "$CROSS_BUILD" ]; then
-		sed -i 's%^my \$vpnc = .*;%my \$vpnc = "/usr/bin/vpnc";%' makeman.pl
+		sed -i 's%^my \$vpnc = .*;%my \$vpnc = "/usr/bin/vpnc";%' src/makeman.pl
```
Comment (ericonr) [inline srcpkgs/vpnc/template:33]: `vsed` ?

File: srcpkgs/vpnc/patches/0001-fix-sys-includes-for-musl.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/vpnc.c b/src/vpnc.c
+index ef1930a..36cae9b 100644
+--- ./src/vpnc.c
++++ ./src/vpnc.c
+@@ -24,7 +24,7 @@
+ #define _GNU_SOURCE
+ #include <assert.h>
+ #include <unistd.h>
+-#include <sys/fcntl.h>
++#include <fcntl.h>
```
Comment (ericonr) [inline srcpkgs/vpnc/patches/0001-fix-sys-includes-for-musl.patch:10]: musl just prints a warning if one does this; do they have `-Werror`?

**Review by ericonr — COMMENTED** [review]
I guess also squash the commits?

Overall LGTM

File: srcpkgs/vpnc/patches/0001-fix-sys-includes-for-musl.patch (hunk 3 above)
Comment (Gottox) [reply srcpkgs/vpnc/patches/0001-fix-sys-includes-for-musl.patch:10]: No, it did at some point during my testing. Removing the patch and offer it upstream instead.

File: srcpkgs/vpnc/template (hunk 2 above)
Comment (Gottox) [reply srcpkgs/vpnc/template:33]: applied

File: srcpkgs/vpnc/template (hunk 1 above)
Comment (Gottox) [reply srcpkgs/vpnc/template:28]: changed.

**Pull request merged**